complex/ft_msg.c

```diff
@@ -542,30 +556,80 @@ int ft_post_recv_bufs(void)
 	return 0;
 }
 
-int ft_recv_msg(void)
+int ft_recv_n_msg(int n)
 {
-	int credits, ret;
+	int credits, ret, recved = 0;
 
+post:
 	if (ft_rx_ctrl.credits > (ft_rx_ctrl.max_credits >> 1)) {
 		ret = ft_post_recv_bufs();
 		if (ret)
 			return ret;
 	}
 
-	credits = ft_rx_ctrl.credits;
-	do {
+	while (recved < n) {
+		credits = ft_rx_ctrl.credits;
 		ret = ft_comp_rx(FT_COMP_TO);
 		if (ret)
 			return ret;
+
+		recved += (ft_rx_ctrl.credits - credits);
```

Comment:
Can we receive more than 1 message here?  If not, I would do recved++.  If so, then the call could receive more than the requested number of messages, and I'd add a brief comment above the function to indicate that.